### ISSUE TITLE
Output invers only possible in free mixer

### ISSUE THREAD

**Issue by EngelMT** [FrSkyRC/ETHOS-Feedback-Community#4740]
A customer and I spent several days working on how to programme a Delta (large Mirage) with 2 x elevators and 2 x ailerons. That sounds simple at first. But our problem was that we had to invert an elevator servo. This always had an effect on the Aileron servo. Because of course we want to use the elevator and aileron flaps together.

So we started using the Wizzard to create a model with 4 Ailerons and without elevator. The system then automatically creates an elevator and an Aileron mixer with 4 outputs each. This works well until you start inverting a servo. Then you run into the problem above.

However, if you realise the same mixer using a free mixer, then you can invert the outputs individually in the mixer menu.

It would be desirable for this to be taken into account when creating such a model in the Wizzard. In other words, to enable the inversion of each output in the mixer menu.

The more I look into it, the more I see the need to create this inversion option for all mixers. I have just tried to tackle our problem again by not selecting an aileron and the V-tail in the Wizzard. Then set 4 outputs in the mixer. But even then there is no option to invert the outputs in the mixer menu. Apart from the fact that you then have to trick again to use the aileron instead of the rudder as the source.

My conclusion is therefore to either provide all mixers with the inversion option or to specify a good real delta mixer.

**Comment by bsongis** [comment]
You should only have to invert the direction at servo level

**Comment by EngelMT** [comment]
If you do it on servo level for excample for Elevator then the Aileron on this output is inverted too. That`s not the solution.

**Comment by bsongis** [comment]
Then you didn't choose the inputs correctly on your Ailerons. This is very important to correctly choose the RIGHT and LEFT ailerons. Because they won't react the same (true on all setups with differential)

**Comment by spoke2570** [comment]
I also would like this. I hate that I have to reverse a servo for ailerons and V-tail rudders to work correctly. Mechanically the ailerons and V-tail rudders work correctly with no reverse as the wings and tails are mirrors of each other. The camber and multi-servo elevators should have the built in mixer reversing. Besides, It is so much nicer to compare mixer and output thermometers when they go the same direction.

**Comment by EngelMT** [comment]
@bsongis If you create a model via the wizzward, you cannot select the channels at first. This is set by the system. Of course you can change it later.

If you convert an existing model to FrSky, then these things are also given. Including the distribution of the servos to the outputs of the RB or Power Box etc. Here, too, you don't actually want to convert the model.

And if, for whatever reason, you have to install a servo in a wing the other way round, then that won't work either. Then you have to be able to change this in the mixer.

Another problem could be if a customer comes from Futaba and uses the SBus servos. These are then addressed and set with Futaba. So also the direction of rotation. Here, too, it must be possible to set the direction of rotation individually in the mixer.

We have perhaps the best and most flexible operating system on the RC market. But we don't have such simple things as a output reversal in the mixer. Unless we use a free mixer. But then the problem arises with the trim. Then I have to set up an auxiliary mixer with the source of the trimmer for both mixers.

Of course, you can manage all this somehow thanks to the flexibility of ETHOS. But is it really necessary to realise such simple things in such a roundabout way?

I'm just imagining what it would be like if a gyro receiver were used!

**Comment by bsongis** [comment]
> @bsongis If you create a model via the wizzward, you cannot select the channels at first. This is set by the system. Of course you can change it later.

This is wrong

![image](https://github.com/user-attachments/assets/619c6ab6-381a-4427-8998-194a531fa83f)

**Comment by bsongis** [comment]
The order of Ailerons is important
- because of the direction
- because of the differential

**Comment by bsongis** [comment]
>  Besides, It is so much nicer to compare mixer and output thermometers when they go the same direction.

~~This is the only reason I would agree with you. But there are other possibilities to solve this problem!~~

But what about the value in microseconds? The Channel value displayed on the bargraph is equal to this value in microseconds!

**Comment by strgaltdel** [comment]
@andreas, i sent you an PM regarding the uni-lua, but added some descriptions regarding this topic.
... among others same screenshot like bertrand.

as bertrand mentioned, it's important to know the "sequence" right-left-right-left in an 4 flap environment

**Comment by Dsrenger** [comment]
I always searched for a graphic or similar how to assign the channels. 
I think neither Ethos itself nor the manual is showing the importance of the assignment. 

I made the convention for my models looking from behind from left to right like I read. 
So left flap is "aileron 1" second from Left "aileron 2" and so on.
Left elevator is "elevator 1" right elevator is "elevator 2"


I always felt a little bit left alone in terms of channel assignments and made my own rule.

Perhaps the user needs a little bit more help, at least through the manual? 
I just had a look... Perhaps I overlook something, but I didn't find anything regarding this.

**Comment by strgaltdel** [comment]
![image](https://github.com/user-attachments/assets/733dbbc9-6ae5-43ec-84a4-3bdd946b2dce)


maybe "Aileron R1","Aileron L1", "Aileron R2", "Aileron L2" would've been clearer ?

**Comment by EngelMT** [comment]
I think I have understood it and it may be that it all works that way. But let's assume that, for whatever reason, a servo has to be installed the other way round like the corresponding servo in the other wing. It will no longer work.

In addition, there is no real explanation in the manual like Dsrenger wrote above. I will contact Wolfgang and Lothar about this. This should definitely be improved.

I am still of the opinion that the option of inverting the outputs in the mixer would solve all similar problems. Then everyone would find a solution to such a problem without having to study the manual.

@bsongis Perhaps you could think again about allowing inversion in the mixer. Or at least allow a trim assignment in the free mixer. Then you could set such a model via free mixers.  We have done this now, but we have also created auxiliary mixers with the trims.

**Comment by Wolepo** [comment]
> ![image](https://private-user-images.githubusercontent.com/20392683/386633774-733dbbc9-6ae5-43ec-84a4-3bdd946b2dce.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2ODI5NjYsIm5iZiI6MTczMTY4MjY2NiwicGF0aCI6Ii8yMDM5MjY4My8zODY2MzM3NzQtNzMzZGJiYzktNmFlNS00M2VjLTg0YTQtM2JkZDk0NmIyZGNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE1VDE0NTc0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlNDNmZWE0NDNlZTE3MjU2YjIwYmFkYzlmMDNkZDAzZDg2OTA5ZWZhMjZkOTAwYjhjMTYzMzY3NzhiMDkyMzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tVsfebjf6VQaxyTB586huxDHcmBymLfT2KxmIwg1-iU)
> 
> maybe "Aileron R1","Aileron L1", "Aileron R2", "Aileron L2" would've been clearer ?

Correct would be QR li 1, QR re 1, QR li 2 and QR re 2

**Comment by bsongis** [comment]
> I think neither Ethos itself nor the manual is showing the importance of the assignment.

I will discuss it with Lothar. Yes it is important, and it may cause issues like Andreas describes.

**Comment by bsongis** [comment]
> I think I have understood it and it may be that it all works that way. But let's assume that, for whatever reason, a servo has to be installed the other way round like the corresponding servo in the other wing. It will no longer work.

If you swap 2 servos physically, you have to swap them in your setup too, but not invert the way the Mix works, or you will break everything and finish the day with an headache.

Trim assignment in the Free Mix is something discussed in another issue. I think it should be done when the Input is a switch. By default it should use the corresponding Trim (with an option to disable it).

I don't see anything exceptional in your setup which would force you to use a Free Mix. Just use the right channels so that the Mix can work as designed!

**Comment by bsongis** [comment]
> maybe "Aileron R1","Aileron L1", "Aileron R2", "Aileron L2" would've been clearer ?

Agreed. Would you mind opening an issue for that?

**Comment by Dsrenger** [comment]
" maybe "Aileron R1","Aileron L1", "Aileron R2", "Aileron L2" would've been clearer ?"


Perhaps I am  a good test object because i really dont know how the assingment should be (And Iam usually not that stupid in these things)

I can think of 3 ways to assign with aileron R1 L1 and so on.

From outer flaps to inner flaps:
![image](https://github.com/user-attachments/assets/343fda58-1b23-47e9-b7fd-35326e6aabaf)

From inner flaps to outer flaps:
![image](https://github.com/user-attachments/assets/48e2a612-9282-4514-8488-20faf6a3a7d3)

from left to right:
![image](https://github.com/user-attachments/assets/7630420d-27c5-42c2-961b-1356b800c4d6)

Or doenst it matter as long as left-right-left-right. I really dont know and I already program 8 flap wings. Guess I was lucky ;)

Looking at it, perhaps the wizards graphics would be one additional place to show the assignments?

**Comment by Dsrenger** [comment]
Just thinking about it a little bit more. 

There is free space in the channel assignment screen. Perhaps these graphics can be shown again here?

Iam just brainstorming, But I guess every konfiguration in the wizard is already an own graphic, so adding the channels as text should be possible.
And potentially reusing them in the assignment screen?

As said, just a quick idea, I am not a programmer, just a user. Perhaps its not so easy with language specific graphics. not sure if they exist today. If not, the idea can be scrapped I think

![image](https://github.com/user-attachments/assets/83483c5e-a746-4e55-b72f-2efdbee71b06)

For sure the Names in the list below should match the names in the graphics.

**Comment by bsongis** [comment]
Excellent idea @Dsrenger bitmaps are often better than a long text!

**Comment by bsongis** [comment]
Would you summarize this request into a new issue? This would be for 1.6.0 I think

**Comment by Dsrenger** [comment]
Need to go to a icehockey match now. Sad, I know

Expect a new issue from me tomorrow morning...

**Comment by strgaltdel** [comment]
@Dennis
2thumbs UP!

@Wolfgang:


> Correct would be QR li 1, QR re 1, QR li 2 and QR re 2

Why do i guess its QR re1, QR li1, QR re2 by using the wizzard:

I.
after finishing the wizzard, push full right and look into channel outputs:
1st column, Ail1, ch1 goes to +50% (standard weighting)
oldschool oTx rule : "what is positive, will go up"
so Aileron1 should be the RIGHT one 
![image](https://github.com/user-attachments/assets/7998fd68-a7eb-4f58-b196-e99913f80259)


II
crosscheck:
apply +50% Differential im Aileron Mixer, push full left
Aileron 1 shows -25%, so it's a flap which must be "underweighted" by pushing left, which is only OK for a RIGHT flap.
![image](https://github.com/user-attachments/assets/be79b6f6-5d5d-4355-9cf6-cb26952c190a)

Both arguments indicate that the wizzard starts (aileron1) with a right flap
This is congruent with the order in the mixers !

i don't want to mess up this issue with this topic,
but wrong infos in your great docu should be prevented
i can mail lothar too, seems he is a little bit "underloaded" on his holiday trip.....
in case you have another thinking,  please pm in Andreas' Forum an we can discuss further

regards
Udo

**Comment by Wolepo** [comment]
I have always counted from left to right and have always been successful. But I admit defeat.

**Comment by yak55** [comment]
Hello, I have always followed the guidelines in the description of the receivers.
Here is a screen of the current TW R-10 / SR 10.

And I think it goes from left to right here too.
I also do it as described for the V-tails.

Greetings Eugen

![IMG_6617](https://github.com/user-attachments/assets/cf982dba-36f6-4807-9af2-d1f4f80937ef)

**Comment by strgaltdel** [comment]
there is no ail-channel assignment shown on the page,
and as said, this has nothing to do with the issue,
would prefer dicussing the topic on engelmt
last words:
my tests were done by configuring the wizzard with a standard rx, so channel assignment may differ to stabi series

**Comment by Wolepo** [comment]
Yes, this is about a delta. Andreas has a delta model, but in practice he moves the elevator from the tailplane to the wings. In my opinion, the delta mixer is wrong in this case and a normal model with 2 X QR and 2 X HR should be programmed. This way, the trim will work correctly and there will be no mutual interference. The direction of travel of the servos can then be freely selected. 
I agree, we should discuss this further in the FrSky forum.
Wolfgang

**Comment by yak55** [comment]
Hi Udo, my understanding of the photo above is that the smaller channel AIL is the left aileron.
The larger channel AIL2 is the right aileron.
Eugen

**Comment by lthole** [comment]
Ch1 is definitely Ail R (QR re). If you select a stabi receiver in the wizard with AETR you get CH1 Ailerons1R, etc.

Then it goes Ail 1R, Ail 2L, Ail 3R, Ail 4L, etc.

![IMG_0216](https://github.com/user-attachments/assets/6d50a10b-0a45-4656-bc23-1b46b14e1980)

**Comment by Wolepo** [comment]
Unfortunately, this is not correct. The channels, even in the outputs, have no page assignment, only the channel numbers and their function. It is up to you how you name them and distribute them over the area. Here is a screenshot created directly with the Wizzard and a gyro receiver.
![Test2](https://github.com/user-attachments/assets/e604dcb1-a07c-4e31-920c-b5d493e1b8c3)
Yours sincerely, Wolfgang

**Comment by lthole** [comment]
Apologies, it’s possible that I renamed them and forgot about it. I couldn’t check because I am travelling at the moment. 

At any rate, Ail1 is definitely Right . Ail2 is Left. Then Ail3 Right, and Ail4 Left.

**Comment by strgaltdel** [comment]
so where we should discuss this ??

There is indeed a contradiction. in the manual of Stabi-Rx series
Even by configurging the wizzard with an SR8, Aileron1 mix output will give positive output by pushing right >> so it indicates a right flap  

the SR manual states, Aileron 1 is channel1, so is the wizzard
![image](https://github.com/user-attachments/assets/c6dcbab3-7c26-40b5-82bf-a8a4d6e37bad)

but then (and this is what @Eugen mentioned before and what could be a fail in the description, dont know)
Ail1 is shown as the left aileron in the manual !
![image](https://github.com/user-attachments/assets/d224bc8c-bada-40ce-a446-3fccc1d55133)

**Comment by lthole** [comment]
I agree this is a contradiction. I ignore the receiver manual and follow Ethos, then just ensure that the auto level corrections go the right way.

**Comment by FlipAirboost** [comment]
Hi all,

And if like Opentx, ethos offered compilation options:

- without pre-programmed mixer for those who like simple and easy things
- with pre-programmed mixer for those who think it's simpler.

because pre-programmed mixers + free mixers become really incomprehensible.

Regards Philippe

**Comment by Dsrenger** [comment]
As requested, I tried to summarize the discussion and a possible solutions in a new issue:

https://github.com/FrSkyRC/ETHOS-Feedback-Community/issues/4753

That said, we highjacked Andreas issue a little bit.

Even when the channel assignment is clear, I recommend to think about adding a possibility to output invers in every mixer as well.
Making clear how the assignment should be is a step to not needing output invers. But there are still edge cases where it is potentially needed.

To summarize, I would like to bring the discussion back to andreas point, as I somehow highjacked his issue.

P.s.: We won 6:2 ;)

**Comment by Wolepo** [comment]
I have opened a new thread in the FrSky forum about the channel sequence in a wing. [See here](https://www.frsky-forum.de/forum/index.php?thread/9777-reihenfolge-der-kan%C3%A4le-in-einer-tragfl%C3%A4che/)

**Comment by bsongis** [comment]
> Hi all,
> 
> And if like Opentx, ethos offered compilation options:
> 
> * without pre-programmed mixer for those who like simple and easy things
> * with pre-programmed mixer for those who think it's simpler.
> 
> because pre-programmed mixers + free mixers become really incomprehensible.
> 
> Regards Philippe

Both are needed!

**Comment by bsongis-frsky** [comment]
As explained above, adding a Direction will only bring more frustration, especially in Mixes providing the Differential feature. The idea of being more explicit inside the wizard is explained in #4753

**Comment by hrenz** [comment]
think this is the correct way if yout hold Ail right
left to rights

![image](https://github.com/user-attachments/assets/30fe059d-ba8a-4d22-82a9-2683e66adae5)

**Comment by Wolepo** [comment]
No, unfortunately that's wrong again.
Although the Wizzard does this, it is not correct. After creating the model with the Wizzard, swap channels 1 and 5 in the output, because CH1/QR1=right and CH5/QR2=left. This is the only way to keep the differentiation positive!

**Comment by bsongis** [comment]
The drawing above seems wrong to me too

**Comment by hrenz** [comment]
The diagram is not wrong, 
it should be like this if you put Ail on the right
But this is not because the assignment of the mixer output to the 2 Ail channels is wrong
Only that would have to be rotated
Your mixer output 1 is the positive output, which should go to K5
Your mixer output 2 is the negative output, which should go to K1
Then everything would be ok

**Comment by bsongis** [comment]
He wants I am fired :)

**Comment by Dsrenger** [comment]
Please not, you are quite valuable, I guess ;)
Fun aside... Thanks for all you do

**Comment by hrenz** [comment]
The MixerOutput assignment to the channels right side , left side must be rotated
MixerOutput 1 to Ail right side K5  Ail2
MixerOutput 2 to Ail left side    K1 Ail 1

![image](https://github.com/user-attachments/assets/907d595b-eca2-4b0d-a1a4-fcde2b87ea66)

![image](https://github.com/user-attachments/assets/49bb1d77-3908-43e8-ba61-eb38da65757f)

![image](https://github.com/user-attachments/assets/0dce5a1f-0e75-46e2-b9dd-f87cf726f640)
